server/room-minor-activity.ts

```diff
@@ -95,7 +96,10 @@ export abstract class MinorActivity {
 			});
 
 			if (!MinorActivityClass) {
-				if (pollData.activityid === 'poll') throw new Error('No minorActivity class provided');
+				// hacky fix to make poll queues work with timers
+				const Poll = pollData.class;
+				// @ts-ignore
+				if (pollData.activityid === 'poll') room.setMinorActivity(new Poll(room, pollData));
```

Comment:
The reason you can't import `Poll` here is that `poll.ts` expects the `Rooms` global to be set up. A less hacky way to solve this is to utilize `require`.
```suggestion
				if (pollData.activityid === 'poll') {
					const {Poll} = require('./chat-plugins/poll');
					room.setMinorActivity(new Poll(room, pollData));
				}
```

Reply:
As far as I know, `Rooms` will always exist by the time we're ending a MinorActivity, so this seems safe to me.

Reply:
let me test it real fast just to make sure. for some reason i though require weren't allowed in TS files. probably my memory just being bad per usual